src/ServiceControl/MessageFailures/Api/RetryMessagesApi.cs

```diff
@@ -0,0 +1,20 @@
+﻿namespace ServiceControl.MessageFailures.Api
+{
+    using System.Threading.Tasks;
+    using Nancy;
+    using NServiceBus;
+    using ServiceControl.CompositeViews.Messages;
+    using ServiceControl.MessageFailures.InternalMessages;
+
+    public class RetryMessagesApi : RoutedApi<string>
+    {
+        public IBus Bus { get; set; }
+
+        protected override Task<Response> LocalQuery(Request request, string input, string instanceId)
+        {
+            Bus.SendLocal<RetryMessage>(m => { m.FailedMessageId = input; });
```

Comment:
This seems odd. We are routing an HTTP request to an endpoint so that it can send itself a message. If we have the instance Id, should we be able to send it the message directly? Just a thought.

Reply:
We originally had it that way. The controller was looking for the instance id and then either did a send or send local. When we introduced the routing API I felt it is good to align to that common API so that multi instace communication always uses the HTTP facade.

Reply:
Another argument would be that the less bus interaction we have and the more we communicate over the API the friendlier it is to rip out things behind the scenes

Reply:
@danielmarbach I like the second argument more. The fact that the instance sends itself a message is an implementation detail. The API is that HTTP requests get routed to the correct instance.